pinot-broker/src/main/java/org/apache/pinot/broker/api/AccessControl.java

```diff
@@ -48,7 +50,7 @@ default boolean hasAccess(RequesterIdentity requesterIdentity) {
    *
    * @return {@code true} if authorized, {@code false} otherwise
    */
-  boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest);
+  AuthorizationResult hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest);
```

Comment:
This is a public stable interface. Directly changing the signature can cause backward incompatibility. You may want to add new APIs and add default impl for the old API

Reply:
I don't think we need to change this one with `BrokerRequest` because it is single-stage engine query.

Reply:
Currently, this method is only used in a single-stage engine. However, according to the Javadoc, it is can also used in columns, permissions, etc. Therefore, I am making it compatible to send messages to the REST controller level.

Sure , will ensure this change is backwards compatible.